packages/svelte-parse/src/main.ts

```diff
@@ -0,0 +1,1219 @@
+import {
+	Point,
+	Node,
+	BaseSvelteTag,
+	SvelteTag,
+	Property,
+	SvelteElement,
+	Directive,
+	Text,
+	Literal,
+	Root,
+	SvelteExpression,
+	VoidBlock,
+	BranchingBlock,
+	Branch,
+	Comment,
+	Parent,
+	SvelteComponent,
+} from 'svast';
+
+import {
+	ParseNodeOptions,
+	ParseOptions,
+	Result,
+	State,
+	SLASH,
+	CLOSE_ANGLE_BRACKET,
+	COLON,
+	EQUALS,
+	QUOTE,
+	APOSTROPHE,
+	TAB,
+	CLOSE_BRACE,
+	BACKTICK,
+	BACKSLASH,
+	AT,
+	OCTOTHERP,
+	RE_BLOCK_BRANCH,
+	RE_SCRIPT_STYLE,
+	RE_COMMENT_START,
+	RE_COMMENT_END,
+	RE_END_TAG_START,
+} from './types_and_things';
+
+import {
+	LINEFEED,
+	OPEN_ANGLE_BRACKET,
+	OPEN_BRACE,
+	UPPERCASE_A,
+	UPPERCASE_Z,
+	LOWERCASE_A,
+	LOWERCASE_Z,
+	SPACE,
+	PIPE,
+} from './types_and_things';
+
+import { void_els } from './void_els';
+
+function is_in_range(number: number, from: number, to: number): boolean {
+	return number >= from && number <= to;
+}
+
+function is_upper_alpha(code: number) {
+	return is_in_range(code, UPPERCASE_A, UPPERCASE_Z);
+}
+
+function is_lower_alpha(code: number) {
+	return is_in_range(code, LOWERCASE_A, LOWERCASE_Z);
+}
+
+function is_void_element(tag_name: string): boolean {
+	return void_els.includes(tag_name);
+}
+
+export function parseNode(opts: ParseNodeOptions): Result | undefined {
+	let index = 0;
+	let quote_type = '';
+	let expr_quote_type = '';
+	let closing_tag_name = '';
+	let brace_count = 0;
+
+	const node_stack: Node[] = [];
+	const state: State[] = [];
+
+	const {
+		value,
+		currentPosition = {
+			line: 1,
+			column: 1,
+			offset: 0,
+		},
+		block = true,
+		childParser,
+		generatePositions = true,
+	} = opts;
+
+	let position = Object.assign(currentPosition, { index });
```

Comment:
You may have to add `ts-ignore` but worth it

```suggestion
	let position = opts.currentPosition || { line: 1, column: 1, offset: 0, index };
	if (opts.currentPosition) position.index = index;
```

nit: `opts.position` maybe ideal?